src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalDao.scala

```diff
@@ -107,6 +109,15 @@ trait BaseByteArrayJournalDao extends JournalDao {
       db.run(actions.transactionally)
     }
 
+  def update(persistenceId: String, sequenceNr: Long, payload: AnyRef): Future[Unit] = {
+    val write = PersistentRepr(payload, sequenceNr, persistenceId)
+    val serializedRow = serializer.serialize(write) match {
+      case Success(t)  => t
+      case Failure(ex) => throw new IllegalArgumentException(s"Failed to serialize ${write.getClass} for update of [$persistenceId] @ [$sequenceNr]")
+    }
+    db.run(queries.update(persistenceId, sequenceNr, serializedRow.message)).map(_ => ())
```

Comment:
Perhaps a `Future[Done]` can be returned, in that case we do not need to use `.map(_ => Done)` at a later point to convert it again.

Reply:
done